MyBus/MyBus/DBManager.swift

```diff
@@ -12,163 +12,189 @@ import RealmSwift
 private let _sharedInstance = DBManager()
 
 public class DBManager: NSObject {
-    let db = try! Realm()
+    var db: Realm?
 
     // MARK: - Instantiation
-    public class var sharedInstance: DBManager
-    {
+    public class var sharedInstance: DBManager {
         return _sharedInstance
     }
 
-    override init() {}
+    override init() {
+        super.init()
+        do {
+            self.db = try Realm()
+        } catch let error as NSError {
+            fatalError("Error opening realm: \(error)")
+        }
+    }
 
     func getCompleteBusRoutes(busLineName: String) -> Results<CompleteBusItineray>? {
-        let results = db.objects(CompleteBusItineray.self).filter("busLineName = '\(busLineName)'")
-        return results
+        if let db = db {
+            let results = db.objects(CompleteBusItineray.self).filter("busLineName = '\(busLineName)'")
+            return results
+        }
+        return nil
     }
 
     func getFavourites() -> List<RoutePoint>? {
-        let users = db.objects(User)
-        if let user = users.first {
-            return user.favourites
+        if let db = db {
```

Comment:
I'm seeing this piece of code repeated in several methods of this class:
`if let db = db { 
    let users = db.objects(User)
    if let user = users.first { 
        //Do something with the user
   }
}`

Perhaps we should have a computed variable "currentUser:User" inside the DBManager class to handle this logic.
 e.g,
`var currentUser:User? = {
  if let db = db { 
    let users = db.objects(User)
    if let user = users.first { 
        //return user
   }
   return nil
}
}`

So that you can in the following sections reference the currentUser variable instead of replicating the logic